docs/icd.md

```diff
@@ -0,0 +1,195 @@
+```
+
+### :space_invader: Authentication
+
+OAuth2 servers will be used to authorize and authenticate users.
+
+Cloud hosting services offer OAuth2, OpenID, and SAML authorization.
+
+### :mag: Audit Logging
+
+An audit log is a record of all API requests. Each entry of an audit log reports the type of the request along with its result, timestamp and client of origin. Audit logs are important for accountability.
+
+Cloud hosting services (such as AWS Cloud Trail) offer auditing tools for an API.
+
+### :vertical_traffic_light: Rate Limiting
+
+In a Denial of Service (DoS) attack, an attacker issues bursts of API requests in an attempt to exhaust finite server resources. If successful, DoS attacks can occupy disk usage, memory, CPU time, and power - slowing or crashing other server programs. This is of particular concern to the Arrow services, which include safety-critical processes.
+
+Cloud hosting services (such as AWS and GCP) offer rate-limiting features to throttle API requests. AWS and GCP can apply throttles across all accounts and clients in a region, an additional defense against distributed DoS (DDoS) attacks originating from many devices.
+
+### :mailbox_with_mail: Endpoints
+
+| Endpoint | Type | Description |
+| ---- | --- |  ---- |
+| `/live` | GET | Returns a boolean `true` - confirms this service is running
+
+## 🤝 Private API - gRPC
+
+A Private API is typically not exposed to the outside world. Private APIs are commonly used for inter-process communication (IPC) between services on the same device.
+
+[`gRPC`](https://grpc.io/) is a modern, open-source Remote Procedure Call (RPC) framework.
+
+Features of gRPC:
++ Serializes data into Protocol Buffers, strongly typed messages
++ Integrated TLS/SSL authentication and encryption
++ Supports both client and server-side streaming.
+
+Arrow services, which are written in Rust Programming Language, access gRPC features through the [`tonic`](https://docs.rs/tonic/latest/tonic/) library.
+
+### :scroll: Files
+
+A `lib-common` repository will contain gRPC interfaces that will be used by every microservice.
+
+File | Description
+---- | -----
+`common-grpc.proto` | Common gRPC interfaces, protocol buffer file.
+`common-grpc.rs` | Autogenerated Rust Code
+
+The `lib-common` crate will be published to a registry, and added as a dependency by each service:
+```toml
+# e.g. svc-scheduler Cargo.toml
+[dependencies]
+lib-common = 0.1.2
+```
+
+Service-specific gRPC files will be defined in each service's repository:
+```tree
+.
+├── Cargo.lock
+├── Cargo.toml # [workspace] members = ['client', 'server']
+├── Makefile
+├── client/
+│   ├── Cargo.lock
+│   ├── Cargo.toml
+│   └── src
+│       ├── grpc.rs # autogenerated during `cargo build`
+│       └── lib.rs
+├── proto/
+│   └── grpc.proto # used to autogenerate server and client stubs
+└── server
+    ├── build.rs
+    ├── Cargo.toml
+    ├── src
+    │   ├── grpc.rs # autogenerated during `cargo build`
```

Comment:
I'm using `make build` which just uses `cargo build` and it appears to execute.

Should be automatic: https://doc.rust-lang.org/cargo/reference/build-scripts.html

Reply:
Interesting. Now `cargo build` works for me after configuring some xcode cli tools..

However,  if I run `make build`, it would say `No Cargo.toml found in any of the subdirectories, skipping cargo check...`.

Reply:
Bizarre